src/jit/flowgraph.cpp

```diff
@@ -17975,12 +17987,19 @@ void                Compiler::fgSetTreeSeqHelper(GenTreePtr tree)
         break;
     }
 
-    fgSetTreeSeqFinish(tree);
+    fgSetTreeSeqFinish(tree, isLIR);
 }
 
 void
-Compiler::fgSetTreeSeqFinish(GenTreePtr tree)
+Compiler::fgSetTreeSeqFinish(GenTreePtr tree, bool isLIR)
 {
+    // If we are sequencing a node that does not appear in LIR,
+    // do not add it to the list.
+    if (isLIR && ((tree->OperGet() == GT_LIST) || tree->OperGet() == GT_ARGPLACE))
+    {
```

Comment:
Shouldn't this just be a check on GTK_NOTLIR? though it looks like you don't have GT_LIST and GT_ARGPLACE in that set for some reason.